.github/workflows/build_test.yml

```diff
@@ -0,0 +1,61 @@
+name: CI
+
+# Controls when the action will run.
+on:
+  push:
+    branches: [ main, '[1-9].[0-9]+-devel' ]
+  pull_request:
+    branches: [ main, '[1-9].[0-9]+-devel' ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build_and_test:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        include:
+        - mode: debug
+          flag: -b
+          define:
+        - mode: debug
+          flag: -b
+          define: >-
+            -DPER_SPECIES_WEIGHT -DPARTICLE_ID -DPHOTONS -DTRIDENT_PHOTONS
```

Comment:
What's the motivation behind this list? Is there a reason we don't include bremsstrahlung for example? What about non-default particle weighting?

Reply:
These are just all the compile-time flags that aren't set by default. They are all enabled here to check that all of those ifdef'ed sections actually compile. No doubt there are still unchecked parts of the code, but this hopefully reduces them. What things we actually choose to enable by default is a whole separate discussion.

Reply:
I thought that one of the non-default weightings was included. Looks like I missed that. Will add it.

Reply:
Also missed bremsstrahlung

Reply:
Done now.